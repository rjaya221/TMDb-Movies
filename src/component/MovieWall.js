import React, {Component} from 'react';

import Photo from './Photo';

class MovieWall extends Component{
    render(){
       
        const moviesList=this.props.moviesList;
       
        return(<div className="photoGrid">
            {moviesList.map((movie)=><div key={movie.id}>
            <figure><Photo url={movie.poster_path}/>
                <figcaption><h5>Title: {movie.title}</h5></figcaption>
                <figcaption><h5>Rating: {movie.vote_average}</h5></figcaption>
                <figcaption><h5>Popularity: {movie.popularity}</h5></figcaption>
                <figcaption><h5>Genre: {movie.genreString}</h5></figcaption>
            </figure>
            </div>)}
        </div>);
    }
}

export default MovieWall;