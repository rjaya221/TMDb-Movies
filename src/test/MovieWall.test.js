import React from 'react';
import { shallow } from '../enzyme';
import MovieWall from '../component/MovieWall';

describe("should rendered MovieWall", () => {
  it("should render my component", () => {
    //let props = [{vote_count: 754, id: 420818, video: false, vote_average: 7.2, title: "The Lion King",poster_path:"/dzBtMocZuJbjLOXvrl4zGYigDzh.jpg"}]
 
   // const wrapper = shallow(<MovieWall moviesList={props}/>);
   // expect(wrapper.getElements()).toMatchSnapshot();
  });
});
