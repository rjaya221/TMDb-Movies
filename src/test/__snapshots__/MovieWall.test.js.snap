// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendered MovieWall should render my component 1`] = `
Array [
  <div
    className="photoGrid"
  >
    <div>
      <figure>
        <Photo
          url="/dzBtMocZuJbjLOXvrl4zGYigDzh.jpg"
        />
        <figcaption>
          <h5>
            Title: 
            The Lion King
          </h5>
        </figcaption>
        <figcaption>
          <h5>
            Rating: 
            7.2
          </h5>
        </figcaption>
        <figcaption>
          <h5>
            Popularity: 
          </h5>
        </figcaption>
        <figcaption>
          <h5>
            Genre: 
          </h5>
        </figcaption>
      </figure>
    </div>
  </div>,
]
`;
